modules/tinymce/src/core/main/ts/bookmark/ResolveBookmark.ts

```diff
@@ -32,27 +32,31 @@ const addBogus = (dom: DOMUtils, node: Node): Node => {
   return node;
 };
 
-const resolveCaretPositionBookmark = (dom: DOMUtils, bookmark: StringPathBookmark) => {
-  const range = dom.createRng();
-  const startPos = CaretBookmark.resolve(dom.getRoot(), bookmark.start);
-  range.setStart(startPos.container(), startPos.offset());
+const resolveCaretPositionBookmark = (dom: DOMUtils, bookmark: StringPathBookmark): Optional<BookmarkResolveResult> => {
+  const startPos = Optional.from(CaretBookmark.resolve(dom.getRoot(), bookmark.start));
+  const endPos = Optional.from(CaretBookmark.resolve(dom.getRoot(), bookmark.end));
+
+  return Optionals.lift2(startPos, endPos, (start, end) => {
```

Comment:
Change here as `CaretBookmark.resolve` can return a nullable value in specific cases and this logic didn't handle that case at all.